packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -4151,6 +4152,43 @@ describe('KeyringController', () => {
         );
       });
     });
+
+    describe('addNewKeyring', () => {
+      it('should call addNewKeyring', async () => {
+        // Mock implementation to avoid having to use real mnemonics.
+        const mockKeyringMetadata: KeyringMetadata = {
+          id: 'mock-id',
+          name: 'mock-keyring',
+        };
+        jest
+          .spyOn(KeyringController.prototype, 'addNewKeyring')
+          .mockImplementationOnce(async () => mockKeyringMetadata);
+
+        await withController(
+          { keyringBuilders: [keyringBuilderFactory(MockKeyring)] },
+          async ({ controller, messenger }) => {
+            const mockKeyringType = KeyringTypes.hd;
```

Comment:
nit: as we are have `MockKeyring` and we are also adding it to builders, perhaps we can use `MockKeyring.type` instead of `KeyringTypes.hd` for this test?